Updated /docs/README.md with linked TOC

joshhus opened the pull request:
TOC done for organization by function and links to GitHub docs.
Where entries are plain text, it's because the target file doesn't have the anchors in them. Will do that Tuesday.

joshhus committed:
Test local TOC

joshhus committed:
Test2

joshhus committed:
fix HL link

joshhus committed:
next round

joshhus committed:
Next

joshhus committed:
Next 2

joshhus committed:
Next 3

joshhus committed:
Next 4

joshhus committed:
Next 5

joshhus committed:
Next 6

joshhus committed:
Next 7

christo4ferris commented:
@joshhus as this is a change to the README.md I will forgo rerunning the Travis build.

The organization is a definite improvement. That said, there are some things that need to be addressed:

1) should reference the use cases being developed by the Requirements WG and the ones cited in this version should be contributed, if they haven't been already.
2) it is inappropriate to reference an IBM-specific sample app (Marbles & Commercial Paper). For now, those references/mentions should be removed. I think that whole last section should be removed and we should just add reference to Requirements WG below the Whitepper WG reference in the TOC.

joshhus commented:
Signed-off-by joshhus

joshhus commented:
What is the specific Travis failure? Details, etc. are non-obvious.

joshhus committed:
Updated per Chris Ferris' feedback:

1) Added Requirements WG link at top.
2) Removed last section on Use cases

Note Binh suggested adding our "old" whitepaper because there isn't a completed WP currently. 
Alternatively, ID can work in the "old" whitepaper contents into the TOC / library, and keep the existing link to Whitepaper WG.

joshhus commented:
@christo4ferris ... I have completed your 1) and 2) fixes above.

christo4ferris commented:
@joshhus thanks, LGTM

srderson commented:
LGTM
